src/PrestaShopBundle/Form/Admin/Type/CustomMoneyType.php

```diff
@@ -44,15 +74,60 @@ public static function getExtendedTypes(): iterable
      */
     public function configureOptions(OptionsResolver $resolver)
     {
+        $defaultCurrencyIso = Currency::getIsoCodeById($this->defaultCurrencyId);
+
         $resolver->setDefaults([
             'precision' => null,
             'scale' => self::PRESTASHOP_DECIMALS,
             'grouping' => false,
             'divisor' => 1,
-            'currency' => 'EUR',
+            'currency' => !empty($defaultCurrencyIso) ? $defaultCurrencyIso : 'EUR',
             'compound' => false,
         ]);
 
         $resolver->setAllowedTypes('scale', 'int');
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        parent::buildView($view, $form, $options);
+
+        $view->vars['money_pattern'] = $this->getFrameworkPattern($options['currency']);
+    }
+
+    /**
+     * Provides currency pattern understandable to symfony, but uses prestashop Locale.
+     *
+     * @param string $currencyCode e.g. EUR, USD
+     *
+     * @return string
+     *
+     * @throws InvalidArgumentException
+     * @throws LocalizationException
+     */
+    private function getFrameworkPattern(string $currencyCode): string
```

Comment:
can this logic be in a separate tested service ?

Reply:
Actually, the first implementation was separated, but it ended up being quite overkill since it was not useful anywhere but here However it complexified some existing interfaces and services for no additional value and induced some breaking change

At least the change here is located to the form type only (which is the only place where it's needed) and it's still unit tested, only it's the form type that is tested directly which is acceptable IMHO

Reply:
I agree that we can keep it as it is, but we need to get rid of the legacy classes (Locale, Price) on the new code.